dapps/kiosk/src/mutations/kiosk.ts

```diff
@@ -80,10 +70,14 @@ export function usePlaceAndListMutation({ onSuccess, onError }: MutationParams)
 
 			if (!cap || !currentAccount?.address) throw new Error('Missing account, kiosk or kiosk cap');
 
+			kioskClient.setSelectedCap(cap);
```

Comment:
I think this is a good indication that this API is slightly wrong: We have a stateful client that we modify here, but it's shared between all components, which breaks most encapsulation principles. Separation of transaction construction and data fetching should help solve this.

Reply:
The overal idea was: 

1. You keep a single `kioskClient`
2. You have a dropdown selector (because a user can choose which kiosk apps put items to), and that sets the client's selectedCap.
3. Whenever you do any TX, it follows that cap.


It's not well depicted on the dapp yet, I was patching it for now. Needs some revamp on the demo.

Reply:
I get the intent but I don’t think it works well to share fetching and transaction construction utilities, which is the root of this problem. I also just don’t love super stateful shared clients like this, I’d much rather that state be managed elsewhere (wherever it makes sense for the app) and then it’s gets used in API calls as needed. In general that approach tends to work better.

Reply:
Similarly we haven’t put network selection logic directly into the base SDK for similar reasons.

Reply:
I refactored it to not keep a stateful `cap`, but instead rely on it on `getOwnerCap` and `ownedKioskTx`.